### PR TITLE
add: export bigquery.js

### LINKED ISSUE #156: script to export event/analytics aggregates to BigQuery
a JS script to export event aggregates to bigquery for selected channels (or all of them)

this is not a priority until #89

### PR DESCRIPTION
**Pull request by samparsky**
Fixes #156

### CODE REVIEW

File: scripts/export-adx.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+const volumeSchema = [
+	{ name: 'id', type: 'STRING', mode: 'REQUIRED' },
+	{ name: 'volume', type: 'FLOAT64', mode: 'REQUIRED' },
+	{ name: 'timestamp', type: 'NUMERIC', mode: 'REQUIRED' }
+]
+
+const priceSchema = [
+	{ name: 'id', type: 'STRING', mode: 'REQUIRED' },
+	{ name: 'price', type: 'FLOAT64', mode: 'REQUIRED' },
+	{ name: 'timestamp', type: 'NUMERIC', mode: 'REQUIRED' }
```
Comment (rori4) [inline scripts/export-adx.js:29]: same here. Use TIMESTAMP

File: scripts/export-stake.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+const { bigQueryTables } = require('../services/constants')
+
+const stakeSchema = [
+	{ name: 'id', type: 'STRING', mode: 'REQUIRED' },
+	{ name: 'owner', type: 'STRING', mode: 'REQUIRED' },
+	{ name: 'amount', type: 'STRING', mode: 'REQUIRED' },
+	{ name: 'poolId', type: 'STRING', mode: 'REQUIRED' },
+	{ name: 'nonce', type: 'STRING', mode: 'REQUIRED' },
+	{ name: 'slashedAtStart', type: 'STRING', mode: 'REQUIRED' },
+	{ name: 'timestamp', type: 'NUMERIC', mode: 'REQUIRED' }
```
Comment (rori4) [inline scripts/export-stake.js:25]: same here. Use TIMESTAMP

File: scripts/index.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+const { BigQuery } = require('@google-cloud/bigquery')
+const logger = require('../services/logger')('bigquery')
+
+const DATASET = process.env.DATASET || 'adex'
+const PROJECT_ID = process.env.PROJECT_ID || 'adex-275614'
+const bigQueryClient = new BigQuery() // missing api key
```
[the remark this answers is not in the document]
Comment (samparsky) [reply scripts/index.js:6]: No, we should always use an environment variable

[the remark this answers is not in the document]
Comment (rori4) [reply scripts/index.js:6]: ok how about like that?
```
const bigQueryClient = new BigQuery({
	keyFilename: process.env.PATH_TO_KEYFILE,
	projectId: PROJECT_ID
})
```

File: scripts/export-analytics.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+const logger = require('../services/logger')('evAggr')
+const { bigQueryTables, collections } = require('../services/constants')
+
+const schema = [
+	{ name: 'channelId', type: 'STRING', mode: 'REQUIRED' },
+	{ name: 'created', type: 'TIMESTAMP', mode: 'REQUIRED' },
+	{ name: 'event_type', type: 'STRING', mode: 'REQUIRED' },
+	{ name: 'earner', type: 'STRING', mode: 'REQUIRED' },
+	{ name: 'count', type: 'STRING', mode: 'REQUIRED' },
+	{ name: 'payout', type: 'STRING', mode: 'REQUIRED' }
```
Comment (rori4) [inline scripts/export-analytics.js:24]: Just talked to @GTitova about this and we need `count` and `payout` to be `NUMERIC`. Can we do that? I am not sure how big the payout number can get here so be careful.

Comment (Ivshti) [reply scripts/export-analytics.js:24]: It should be ok @samparsky, as we don't need absolute precision in the analytics

Comment (samparsky) [reply scripts/export-analytics.js]: Yeah I have made the change

File: scripts/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (samparsky) [reply scripts/index.js]: Bigquery lib already has a env var name that fit's the purpose `GOOGLE_APPLICATION_CREDENTIALS`

**Pull request merged**